include/roboteam_ai/world/Field.h

```diff
@@ -33,6 +33,20 @@ struct FieldArc {
  * - The location and sizes of all arcs on the field. <br>
  * - Important and frequently used locations of the field, e.g. positions around our and the opponents goal.
  *
+ * Some invariants regarding to the field are:
+ * - The left side of the field always corresponds with our side of the field (where our goal is located) and the right side of the field always corresponds with the opponents side
+ * of the field (where the opponents goal is located). So our goal will always have the lowest x-coordinate and the opponents goal will always have the highest x-coordinate.
+ * - The center position of the field has (0, 0) as coordinate, which means that every location is indicated by the relative difference towards the center of the field. And that
+ * every position is mirrored with regard to the center position, e.g. the center of our goal has a coordinate of the form (-x, 0) and the center of the opponents goal has a
+ * coordinate of the form (x, 0). Another example is that the bottommost y-coordinate of the field is of the form -y where the topmost y-coordinate of the field is of the form y.
+ * - The left side of the field corresponds with the lowest x-coordinate (is a negative value), whereas the right side of the field corresponds with the highest y-coordinate (is a
+ * positive value). The bottom side of the field corresponds with the lowest y-coordinate (is a negative value), whereas the top side of the field corresponds to the highest
+ * y-coordinate (is a positive value).
+ * - Every horizontal line of the field stored as FieldLineSegment has their begin position equal to the leftmost position of that line (position with the lowest x-coordinate) and
+ * their end position equal to the rightmost position of that line (position with highest x-coordinate). Every vertical line of the field stored as FieldLineSegment has their begin
+ * position equal to the bottommost position of that line (position with the lowest y-coordinate) and ther end position equal to the topmost position of that line (position with
+ * the highest y-coordinate).
+ *
```

Comment:
This is far too much. If you need this amount of documentation either your code is not readable enough, or your documentation is to detailed. (i gues the second). just make nice function names that say what they do and it is fine.

Reply:
These invariants are really needed. I did not understand these things at first sight, so I added them because if I do not understand them then others are also more likely to not understand them either. And a lot of functions outside the Field class do computations based on these assumptions.

Reply:
I see. It is indeed basic knowledge, which if you don't know it then you cannot work on this codebase. However, this can be more easily explained with some kind of drawing then. for example something like this, but bigger:

```
(-x,y)_________________________ (x,y)
|___                              |
|    |                            |
|    |           (0,0)            | (their side)
| __ |                            | (rightFieldLine)
|                                 |
(-x, -y)_______________________(x, -y)
              (bottomFieldLine)
```

Reply:
or make it shorter: 

About coordinates:
The center of the field is (0,0), therefore the left bottom corner is -x,-y and the top right corner is x,y. In roboteam_ai we our side is always the left side of the field. If we play on the right side in real-life the incoming field messages are rotated when they are received.

Reply:
Well I like the drawing, but the invariant for the FieldLine remains, because it is not explained by this drawing.